controllers/awsmachine_controller.go

```diff
@@ -673,6 +731,38 @@ func (r *AWSMachineReconciler) resolveUserData(machineScope *scope.MachineScope,
 	return encryptedCloudInit, nil
 }
 
+func (r *AWSMachineReconciler) ignitionUserData(scope *scope.MachineScope, objectStoreSvc services.ObjectStoreInterface, userData []byte) ([]byte, error) {
+	if objectStoreSvc == nil {
+		return nil, errors.New("object store service not available")
+	}
+
+	objectURL, err := objectStoreSvc.Create(scope, userData)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating userdata object")
+	}
+
+	ignData := &ignTypes.Config{
+		Ignition: ignTypes.Ignition{
+			Version: "2.3.0",
```

Comment:
Does this relate to the ignition version (and `DefaultIgnitionversion`)? If it does will it still work if he user supplies there own version?

(sorry i don't know enough about ignition.....will have a look)

Reply:
>Does this relate to the ignition version (and DefaultIgnitionversion)?

Yes, it does.

>If it does will it still work if he user supplies there own version?

This is hardcoded, as right now version 2.3 is the only supported/valid version (Similar to what we've done in CABPK). 

Once version is made flexible, the implementation here should be expanded.

Reply:
Ok cool....and thanks for the explanation :+1: